### ISSUE TITLE
REMOVE_DUPLICATE_ISO_VERTS in smvv.f90

### ISSUE THREAD

**Issue by marcosvanella** [firemodels/fds#2632]
This one is for Glenn:

I'm running the /Verification/Visualization/plume5c.fds test case, compiling with gnu and intel.
It so happens that routine

REMOVE_DUPLICATE_ISO_VERTS(VERTS,NVERTS,TRIANGLES,NTRIANGLES) (line ~120 smvv.f90) 

gets called with a size for VERTS 3 times NVERTS, or 5376 in the test case. This routine call within a loop DO IFROM=2, NVERTS a function

IMATCH = GET_MATCH(IFROM,VERTS,ITO-1)

where ITO is ~IFROM-1. The definitions on the function are (line ~100 on smvv.f90):

INTEGER FUNCTION GET_MATCH(IFROM,VERTS,NVERTS)

  INTEGER, INTENT(IN) :: NVERTS
  INTEGER, INTENT(IN) :: IFROM
  REAL(FB), INTENT(IN), DIMENSION(3*NVERTS+3), TARGET :: VERTS

So, in this function NVERTS=ITO-1 when called. Using the intel compiler VERTS is still defined of size 5376 (who knows why), while with gnu is assumed of size 3*(ITO-1)+3. After 4 calls of this function the bounds of the pointer array VERTS are overrrun in GET_MATCH for gnu (out of bounds error), while intel gracefully keeps doing its thing. 
Shouldn't the size of VERTS in GET_MATCH be defined as the actual total allocated size 5376 here?

Thank you!

**Comment by mcgratta** [comment]
Marcos, just click on "Assignee" at right to assign the issue to Glenn or whomever.

**Comment by marcosvanella** [comment]
Also, there are two nullify/allocate statements on pointer arrays XYZVERTS, TRIANGLES that seem to produce a small memory leak (see lines ~196-198 of smvv.f90):

==28103== 12,000 bytes in 1 blocks are definitely lost in loss record 319 of 430
==28103==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==28103==    by 0x4B6DF2: **isosmoke_MOD_iso_to_geom (smvv.f90:196)
==28103==    by 0x4B9B6D: __isosmoke_MOD_iso_to_file (smvv.f90:50)
==28103==    by 0xB86B75: __dump_MOD_dump_isof (dump.f90:3444)
==28103==    by 0xC0779D: __dump_MOD_dump_mesh_outputs (dump.f90:138)
==28103==    by 0xFC8F77: MAIN** (main.f90:969)
==28103==    by 0x1076789: main (main.f90:5)
==28103== 
==28103== 12,000 bytes in 1 blocks are definitely lost in loss record 320 of 430
==28103==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==28103==    by 0x4B6F8D: **isosmoke_MOD_iso_to_geom (smvv.f90:198)
==28103==    by 0x4B9B6D: __isosmoke_MOD_iso_to_file (smvv.f90:50)
==28103==    by 0xB86B75: __dump_MOD_dump_isof (dump.f90:3444)
==28103==    by 0xC0779D: __dump_MOD_dump_mesh_outputs (dump.f90:138)
==28103==    by 0xFC8F77: MAIN** (main.f90:969)
==28103==    by 0x1076789: main (main.f90:5)

 Looks like passing the 

```
DEALLOCATE(XYZVERTS) ! these variables were allocated in ISO_TO_GEOM

DEALLOCATE(TRIANGLES)
```

statements of lines ~57-59 in smvv.f90 outside of the 

IF (NTRIANGLES>0.AND.NXYZVERTS>0) THEN
...
ENDIF

conditional, takes care of this.

**Comment by gforney** [comment]
should be fixed in commit 98ce039 .  I made some subtle mistakes that valgrind (and marcos!) caught .  I'll have to try this tool in smokeview

**Comment by marcosvanella** [comment]
Issue closed. Glenn, there is a gui viewer for valgrind, called valkyrie. I'm installing it on my mac, to check it out.
